### PR TITLE
Use std::integer_sequence instead of own IntegerSequence

### PR DESCRIPTION
**Pull request by BenjaminW3**

### CODE REVIEW

**Comment by BenjaminW3** [comment]
Ready to be approved/merged!

File: include/alpaka/vec/Traits.hpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include <alpaka/extent/Traits.hpp>
 #include <alpaka/offset/Traits.hpp>
 #include <alpaka/idx/Traits.hpp>
+#include <alpaka/meta/IntegerSequence.hpp>
```
Comment (ax3l) [inline include/alpaka/vec/Traits.hpp:17]: this should probably read
```suggestion
#include <utility>
```

File: include/alpaka/vec/Vec.hpp [hunk 2]
```diff
@@ -59,7 +59,7 @@ namespace alpaka
             typename TIdxSize,
             TIdxSize... TIndices>
         ALPAKA_FN_HOST_ACC auto createVecFromIndexedFnArbitrary(
-            meta::IntegerSequence<TIdxSize, TIndices...> const & indices,
+            std::integer_sequence<TIdxSize, TIndices...> const & indices,
```
Comment (ax3l) [inline include/alpaka/vec/Vec.hpp:63]: this file should include `<utility>`

File: test/unit/vec/src/VecTest.cpp [hunk 3]
```diff
@@ -39,7 +39,7 @@ TEST_CASE("basicVecTraits", "[vec]")
     // alpaka::vec::subVecFromIndices
     {
         using IdxSequence =
-            alpaka::meta::IntegerSequence<
+            std::integer_sequence<
```
Comment (ax3l) [inline test/unit/vec/src/VecTest.cpp:44]: this file should include `<utility>`

File: include/alpaka/vec/Traits.hpp (hunk 1 above)
Comment (BenjaminW3) [reply include/alpaka/vec/Traits.hpp:17]: We still use some methods from `IntegerSequence.hpp`

**Comment by BenjaminW3** [comment]
All the comments have been incorporated. Please approve.

**Review by ax3l — APPROVED** [review]

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**